plugins/callback/diy.py

```diff
@@ -0,0 +1,1421 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Trevor Highfill <trevor.highfill@outlook.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+  callback: diy
+  callback_type: stdout
+  short_description: Customize the output
+  description:
+    - Callback plugin for the DIYer, because it allows you to supply your own custom callback templates to be output
+  version_added: "2.10"
+  author: Trevor Highfill (@theque5t)
+  extends_documentation_fragment:
+    - default_callback
+  notes:
+    - Uses the L(default,./default.html) callback plugin output when a custom callback message(C(msg)) is not provided.
+    - Makes the callback event data available via the C(ansible_callback_diy) dictionary, which can be used in the templating context for the options.
```

Comment:
Option names are typeset with `I(...)`, values with `C(...)`.
```suggestion
    - Makes the callback event data available via the I(ansible_callback_diy) dictionary, which can be used in the templating context for the options.
```

Reply:
@felixfontein My apologies if I'm misunderstanding something, but technically `ansible_callback_diy` is not an option name. It is a dictionary name that can be used as part of a value that would be supplied to an option. Therefore, I think it should stay typeset with `C(...)`? Or, perhaps I'm not understanding the scope of an "option name"?

Reply:
Ok, I think now I understand. Use `C(...)` in that case.

Reply:
Excellent, will do.